nncf/common/quantization/quantizer_propagation/solver.py

```diff
@@ -0,0 +1,1211 @@
+        for pq, qp_id in self._prop_quantizer_vs_quantization_point_id.items():
+            if qp_id not in final_quantizer_setup.quantization_points:
+                self._quant_prop_graph.remove_propagating_quantizer(pq)
+            else:
+                final_qconfig = final_quantizer_setup.quantization_points[qp_id].qconfig
+                if strict:
+                    def is_final_qconfig_compatible_to_initial(initial_qconfig: QuantizerConfig):
+                        return final_qconfig.per_channel == initial_qconfig.per_channel and \
+                               final_qconfig.mode == initial_qconfig.mode and \
+                               final_qconfig.num_bits == initial_qconfig.num_bits and \
+                               (final_qconfig.signedness_to_force == initial_qconfig.signedness_to_force or
+                                initial_qconfig.signedness_to_force is None or
+                                final_qconfig.signedness_to_force is None)
+
+                    compatible_initial_qconfs = list(
+                        filter(is_final_qconfig_compatible_to_initial,
+                               self.quantizer_setup.quantization_points[qp_id].possible_qconfigs))
+                    if not compatible_initial_qconfs:
+                        raise RuntimeError("The final quantizer setup has configurations that were not present in the "
+                                           "initial proposal!")
+                    if final_qconfig.signedness_to_force is None:
+                        initial_qconfs_signedness_values = {qc.signedness_to_force for qc in compatible_initial_qconfs}
+                        if None not in initial_qconfs_signedness_values and len(initial_qconfs_signedness_values) == 1:
+                            # The initial configs were either all forced-signed or all forced-unsigned - should set
+                            # final qconfig's forced field appropriately
+                            final_qconfig.signedness_to_force = initial_qconfs_signedness_values.pop()
+
+                pq.potential_quant_configs = [final_qconfig]
+        return FinalizedQuantizationProposal(final_quantizer_setup,
+                                             self._quant_prop_graph)
+
+
+class QuantizerPropagationSolver:
+    """
+    Analyzes a fresh QuantizerPropagationStateGraph object according to HW
+    configuration supplied in the initializer and produces the list of insertion
+    commands that correspond to the final state of the quantizer propagation graph
+    when the model has the most contol flow graph edges quantized according to HW
+    capabilities.
+    """
+
+    DEFAULT_QUANTIZATION_TYPES = [QuantizerConfig(
+        num_bits=8,
+        mode=QuantizationMode.SYMMETRIC,
+        signedness_to_force=None,
+        per_channel=False)]
+
+    DEFAULT_PROPAGATION_STRATEGY = PropagationStrategy.MERGE_WITH_SINGLE_FQ_RESULT
+
+    def __init__(self,
```

Comment:
Could you please provide a docstring for this method?

Reply:
Done